ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Wiki/WikiAttachmentsSample.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.TeamFoundation.Wiki.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.TeamServices.Samples.Client.Wiki
+{
+    [ClientSample(WikiConstants.AreaName, WikiConstants.AttachmentsResourceName)]
+    public class WikiAttachmentsSample : ClientSample
+    {
+        [ClientSampleMethod]
+        public WikiAttachmentResponse AddAttachment()
+        {
+            VssConnection connection = this.Context.Connection;
+            WikiHttpClient wikiClient = connection.GetClient<WikiHttpClient>();
+
+            Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync(projectId).SyncResult();
+
+            if (wikis == null || wikis.Count == 0)
+            {
+                Console.WriteLine("No wikis exist to create wiki page");
+
+                return null;
+            }
+
+            WikiV2 wiki = wikis[0];
```

Comment:
You might consider a helper method (in a Wiki util class, for example) that return the same dummy/sample wiki for a given project (and creates if it necessary). This reduces the redundant code in most of your methods (e.g. if (wikis == null ....) and ensures the sample always works (assuming there is a project in the account the user attempted to run your samples against).

```
Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
WikiV2 wiki = WikiSampleHelpers.FindOrCreateCodeWiki(projectId);
```